// element-plus@1.2.0-beta.5/lib/hooks/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var index = require('./use-attrs/index.js');
var index$1 = require('./use-focus/index.js');
var index$2 = require('./use-form-item/index.js');
var index$3 = require('./use-global-config/index.js');
var index$4 = require('./use-locale/index.js');
var index$5 = require('./use-lockscreen/index.js');
var index$6 = require('./use-modal/index.js');
var index$7 = require('./use-model-toggle/index.js');
var index$8 = require('./use-popper/index.js');
var index$9 = require('./use-prevent-global/index.js');
var index$a = require('./use-restore-active/index.js');
var index$b = require('./use-same-target/index.js');
var index$c = require('./use-teleport/index.js');
var index$d = require('./use-throttle-render/index.js');
var index$e = require('./use-timeout/index.js');



exports.useAttrs = index.useAttrs;
exports.useFocus = index$1.useFocus;
exports.useFormItem = index$2.useFormItem;
exports.useFormItemProps = index$2.useFormItemProps;
exports.useGlobalConfig = index$3.useGlobalConfig;
exports.buildTranslator = index$4.buildTranslator;
exports.localeContextKey = index$4.localeContextKey;
exports.localeProviderMaker = index$4.localeProviderMaker;
exports.provideLocale = index$4.provideLocale;
exports.translate = index$4.translate;
exports.useLocale = index$4.useLocale;
exports.useLocaleProps = index$4.useLocaleProps;
exports.useLockscreen = index$5.useLockscreen;
exports.useModal = index$6.useModal;
exports.useModelToggle = index$7.useModelToggle;
exports.useModelToggleEmits = index$7.useModelToggleEmits;
exports.useModelToggleProps = index$7.useModelToggleProps;
exports.DARK_EFFECT = index$8.DARK_EFFECT;
exports.LIGHT_EFFECT = index$8.LIGHT_EFFECT;
exports.usePopperControlProps = index$8.usePopperControlProps;
exports.usePopperHook = index$8.usePopperHook;
exports.usePopperProps = index$8.usePopperProps;
exports.usePreventGlobal = index$9.usePreventGlobal;
exports.useRestoreActive = index$a.useRestoreActive;
exports.useSameTarget = index$b.useSameTarget;
exports.useTeleport = index$c.useTeleport;
exports.useThrottleRender = index$d.useThrottleRender;
exports.useTimeout = index$e.useTimeout;
